### PR TITLE
Emscripten update

### PR DESCRIPTION
**Pull request by Ghabry**

### CODE REVIEW

**Comment by Ghabry** [comment]
Invented a new icudata install hack that works:
The problem is that emscripten is not detected as a crosscompile but we must crosscompile because the tools don't work.
The new function "icu_force_data_install" will force a non-assembly data file build and install the file.
Not nice but works...

File: emscripten/1_download_library.sh [hunk 1]
```diff
@@ -16,22 +16,26 @@ msg " [1] Preparing Emscripten SDK"
...
 
 # Fetch the latest registry of available tools.
-./emsdk update
+./emsdk update-tags
 
 # Download and install the latest SDK tools and set up the compiler configuration to point to it.
-./emsdk install sdk-incoming-64bit
-./emsdk activate sdk-incoming-64bit
+./emsdk install latest-upstream
+./emsdk activate latest-upstream
```
Comment (Ghabry) [inline emscripten/1_download_library.sh]: I wanted to use 1.38.46 (latest release) here but when I do this libpng can't find "pow"

**Comment by Ghabry** [comment]
Seems that since v1.39.0 emscripten uses upstream wasm clang instead of fastcomp. Will update this and try again.
In general I suggest pinning to a specific emscripten version to prevent problems when recompiling because of breaking changes like e.g.
v1.39.1:
Only MEMFS is included by default, others (NODEFS, IDBFS, WORKERFS, PROXYFS)
   must be linked in explicitly, using `-lnodefs.js`, `-lidbfs.js`',
   `-lworkerfs.js`, `-lproxyfs.js`. See #9645

**Comment by Ghabry** [comment]
Pinned this now to 1.39.5 (latest version right now) to prevent later breakage when rebuilding.

Player test still pending but libraries built fine.

File: emscripten/1_download_library.sh [hunk 2]
```diff
@@ -16,22 +16,26 @@ msg " [1] Preparing Emscripten SDK"
...
 
 # Set the current Emscripten path
 # The following line fails to run in jenkins jobs. Running inner called scripts instead works.
-#. ./emsdk_env.sh
 ./emsdk construct_env
 . ./emsdk_set_env.sh
 
+# Activating emsdk overwrites a config file in the home directory
+# by copying it to emsdk the local file is used
+cp ~/.emscripten .
```
Comment (carstene1ns) [inline emscripten/1_download_library.sh]: I do not like this. Need to check, if there is a better way.

Comment (Ghabry) [reply emscripten/1_download_library.sh]: I found a better solution that involves creating of an empty file in the emsdk directory

File: emscripten/1_download_library.sh [hunk 3]
```diff
@@ -16,19 +16,23 @@ msg " [1] Preparing Emscripten SDK"
...
+./emsdk update-tags
 
 # Download and install the latest SDK tools and set up the compiler configuration to point to it.
-./emsdk install sdk-incoming-64bit
-./emsdk activate sdk-incoming-64bit
+./emsdk install 1.39.5
+./emsdk activate 1.39.5
 
 # Set the current Emscripten path
 # The following line fails to run in jenkins jobs. Running inner called scripts instead works.
```
Comment (carstene1ns) [inline emscripten/1_download_library.sh]: This comment is wrong now.

Comment (Ghabry) [reply emscripten/1_download_library.sh]: I just saw in the emsdk install output that I shall use "source ./emsdk_env.sh" to setup the environment. Changed this now (and removed the comment).

Comment (carstene1ns) [reply emscripten/1_download_library.sh]: Well, I hope that this works on jenkins. The comment was made, because *exactly this* did not work.

**Comment by carstene1ns** [comment]
Toolchain on jenkins updated.

**Pull request merged**